mne/viz/eyetracking/heatmap.py

```diff
@@ -59,31 +64,76 @@ def plot_gaze(
     from mne import BaseEpochs
     from mne._fiff.pick import _picks_to_idx
 
+    from ...preprocessing.eyetracking.utils import (
+        _check_calibration,
+        get_screen_visual_angle,
+    )
+
     _validate_type(epochs, BaseEpochs, "epochs")
     _validate_type(alpha, "numeric", "alpha")
     _validate_type(sigma, ("numeric", None), "sigma")
-    width = _ensure_int(width, "width")
-    height = _ensure_int(height, "height")
 
+    # Get the gaze data
     pos_picks = _picks_to_idx(epochs.info, "eyegaze")
     gaze_data = epochs.get_data(picks=pos_picks)
     gaze_ch_loc = np.array([epochs.info["chs"][idx]["loc"] for idx in pos_picks])
     x_data = gaze_data[:, np.where(gaze_ch_loc[:, 4] == -1)[0], :]
     y_data = gaze_data[:, np.where(gaze_ch_loc[:, 4] == 1)[0], :]
+    unit = epochs.info["chs"][pos_picks[0]]["unit"]  # assumes all units are the same
 
     if x_data.shape[1] > 1:  # binocular recording. Average across eyes
         logger.info("Detected binocular recording. Averaging positions across eyes.")
         x_data = np.nanmean(x_data, axis=1)  # shape (n_epochs, n_samples)
         y_data = np.nanmean(y_data, axis=1)
     canvas = np.vstack((x_data.flatten(), y_data.flatten()))  # shape (2, n_samples)
 
+    # Check that we have the right inputs
+    if calibration is not None:
+        if width is not None or height is not None:
```

Comment:
How do folks feel about deprecating the width and height parameters of plot_gaze, in favor of only relying on the calibration instance for this info? It could simplify the API/code, and in my mind this is akin to passing in a montage object for functions that require channel locations for MEEG data.

Reply:
I guess we could. I'm not sure how widely used these are currently, especially since we only support reading EyeLink data currently, which should (always?) have a calibration instance. But if there are cases where people might have eye data without a Calibration instance, being able to pass width/height can be useful. But this could be ameliorated by making `Calibration` easy enough to construct by the user, too...

Reply:
Ah, that's a very good point.. I'm actually working with some open-access eyelink data right now that are shared as `.mat` files, so don't have the calibration info.